packages/ibm-products/src/components/TagOverflow/TagOverflowModal.js

```diff
@@ -104,11 +105,16 @@ export const TagOverflowModal = ({
         />
       </ModalHeader>
       <ModalBody className={`${blockClass}__body`} hasForm>
-        {getFilteredItems().map(({ label, id, ...other }) => (
-          <Tag {...other} filter={false} key={id}>
-            {label}
-          </Tag>
-        ))}
+        {getFilteredItems().map(({ label, id, filter, ...other }) => {
+          if (overflowType !== 'tag') {
+            filter = false;
```

Comment:
we should never be mutating data like this. the conditional should be set as a variable (preferably) or done in the prop
```
const isFilterable = overflowType !== 'tag';
filter={isFilterable}
// OR
filter={overflowType !== 'tag'}
```

Reply:
Done